Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.swift

```diff
@@ -0,0 +1,182 @@
+                    }
+                    .padding(6)
+                    .background(.thinMaterial)
+                    .cornerRadius(10)
+                }
+            }
+            .alert(
+                "Select terminal",
+                isPresented: $model.terminalSelectorIsOpen,
+                actions: { terminalPickerButtons }
+            )
+        }
+    }
+    
+    /// Buttons for each the available terminals on the last added utility element.
+    @ViewBuilder var terminalPickerButtons: some View {
+        ForEach(model.lastAddedElement?.assetType.terminalConfiguration?.terminals ?? []) { terminal in
+            Button(terminal.name) {
+                model.lastAddedElement?.terminal = terminal
+                model.terminalSelectorIsOpen = false
+                model.addTerminal(to: lastSingleTap!.mapPoint)
+            }
+        }
+    }
+    
+    @ViewBuilder var configurationView: some View {
+        Form {
+            Section {
+                List(model.filterBarrierCategories, id: \.name) { category in
+                    HStack {
+                        Text(category.name)
+                        Spacer()
+                        if category === model.selectedCategory {
+                            Image(systemName: "checkmark")
+                                .foregroundColor(.accentColor)
+                        }
+                    }
+                    // Allows the whole row to be tapped. Without this only the text is
+                    // tappable.
+                    .contentShape(Rectangle())
+                    .onTapGesture {
+                        if category.name == model.previousCategory?.name {
+                            model.unselectCategory(category)
+                        } else {
+                            model.selectCategory(category)
+                        }
+                    }
+                }
+            } header: {
+                Text("Category")
+            } footer: {
+                Text("Choose a category to run the valve isolation trace. The selected utility category defines constraints and conditions based upon specific characteristics of asset types in the utility network.")
+            }
+            Section {
+                Toggle(isOn: $model.includesIsolatedFeatures) {
+                    Text("Include Isolated Features")
+                }
+            } header: {
+                Text("Other Options")
+            } footer: {
+                Text("Choose whether or not the trace should include isolated features. This means that isolated features are included in the trace results when used in conjunction with an isolation trace.")
+            }
+            .toggleStyle(.switch)
+        }
+        .navigationTitle("Configuration")
+        .navigationBarTitleDisplayMode(.inline)
+        .toolbar {
+            ToolbarItem(placement: .confirmationAction) {
+                Button("Done") { isConfigurationPresented.toggle() }
```

Comment:
`isConfigurationPresented` shouldn't ever be `false` when the button is tapped, but setting it to `false` in the button's actions makes it absolutely clear that the presented view should go away.

```suggestion
                Button("Done") { isConfigurationPresented = false }
```